### PR TITLE
reverse sign on pause time

### PR DESCRIPTION
**Pull request by patrickweh**

### CODE REVIEW

File: database/migrations/2024_10_20_114910_reverse_sign_for_pause_work_times_on_work_times_table.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        Schema::table('work_times', function (Blueprint $table) {
+            $table->bigInteger('total_time_ms')->default(0)->change();
+        });
+
+        DB::statement('UPDATE work_times SET total_time_ms = total_time_ms * -1 WHERE is_pause = true');
+    }
+
+    public function down(): void
+    {
+        DB::statement('UPDATE work_times SET total_time_ms = total_time_ms * -1 WHERE is_pause = true');
```
Comment (SirSplasch) [inline database/migrations/2024_10_20_114910_reverse_sign_for_pause_work_times_on_work_times_table.php:21]: use ABS(total_time_ms)

**Review by SirSplasch — CHANGES_REQUESTED** [review]
Why not use bcmath on total_time_ms calculation?

**Comment by patrickweh** [comment]
> Why not use bcmath on total_time_ms calculation?

total time is integer only, no need to use bcmath

File: src/Actions/WorkTime/UpdateLockedWorkTime.php [hunk 2]
```diff
@@ -37,6 +37,10 @@ public function performAction(): Model
             } else {
                 $workTime->total_time_ms = Carbon::parse($this->data['ended_at'])
                     ->diffInMilliseconds(Carbon::parse($workTime->started_at)) - $workTime->paused_time_ms;
```
Comment (SirSplasch) [inline src/Actions/WorkTime/UpdateLockedWorkTime.php:39]: bcsub

**Review by SirSplasch — CHANGES_REQUESTED** [review]

File: src/Actions/WorkTime/UpdateLockedWorkTime.php [hunk 3]
```diff
@@ -35,8 +35,15 @@ public function performAction(): Model
                 $workTime->total_time_ms = 0;
                 $workTime->is_locked = false;
             } else {
-                $workTime->total_time_ms = Carbon::parse($this->data['ended_at'])
-                    ->diffInMilliseconds(Carbon::parse($workTime->started_at)) - $workTime->paused_time_ms;
+                $workTime->total_time_ms = bcsub(
+                    Carbon::parse($this->data['ended_at'])->diffInMilliseconds(Carbon::parse($workTime->started_at)),
+                    $workTime->paused_time_ms,
+                0
```
Comment (SirSplasch) [inline src/Actions/WorkTime/UpdateLockedWorkTime.php:41]: indent

**Review by SirSplasch — CHANGES_REQUESTED** [review]

**Review by SirSplasch — APPROVED** [review]

**Pull request merged**